recipes/thrust/all/conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, tools
+import os
+
+class ThrustConan(ConanFile):
+    name = "thrust"
+    license = "Apache-2.0"
+    description = ("Thrust is a parallel algorithms library which resembles"
+                   "the C++ Standard Template Library (STL).")
+    topics = ("parallel", "stl", "header-only")
+    homepage = "https://thrust.github.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    no_copy_source = True
+    options = {"thrust_device_system": ["cuda", "cpp", "omp", "tbb"]}
+    default_options = {"thrust_device_system": "cuda"}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.develop:
+            self.options.thrust_device_system = "cpp"
```

Comment:
It's to build the package for `cpp` so that the test_package doesn't need any additional package.  By default the package is built for `CUDA`. If I don't do this, how I can use the package with `self.options.thrust_device_system = XX` only for the test_package?

Reply:
Check the link, you can define options of your requirements (`test_package/conanfile.py` has an implicit requirement to the recipe it is testing):

![image](https://user-images.githubusercontent.com/1406456/80004573-c7dea300-84c2-11ea-82ff-a0ab68c52dca.png)

Reply:
I've done that, but for some reason it only works  with `cpp` from the test_package. If I try `tbb` it fails, because it doesn't add `tbb` as a requirement, though I already added the `requires` depending on `self.options.device_system` inn the main package. I'll review it.